### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by RobertAcsinte**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -5,10 +5,12 @@ import "./App.css"
 import { Routes, Route } from 'react-router-dom';
 import WizkidDetails from "./pages/WizkidDetails";
 import WizkidEdit from "./pages/WizkidEdit";
+import Login from "./pages/Login";
+import { useState } from "react";
 
 function App() {
 
-  const {fetchWizkids} = useContext(WizkidsContext);
+  const {fetchWizkids, token} = useContext(WizkidsContext);
```
Comment (cjbreg) [inline src/App.js]: Might be interesting/useful to create a separate context for authentication stuff, like `token`.

File: src/components/Navbar.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+import { useNavigate } from "react-router-dom";
 
 function Navbar() {
+  let navigate = useNavigate();
+
   return (
     <nav>
       <a className='company-logo-navbar' href="/">OWOW</a>
-      <a className='login-text-navbar' href="/">Login</a>
+      <a className='login-text-navbar' onClick={() => navigate("/login")}>Login</a>
```
Comment (cjbreg) [inline src/components/Navbar.js]: You could make this button a separate component that could change based on the authentication state.
Login if token is null and signout if token is available for example

File: src/pages/WizkidEdit.js [hunk 3]
```diff
@@ -30,30 +31,47 @@ function WizkidEdit() {
...
             <button className="wizkid-edit-button-edit">
                 Save
             </button>
             {error && <p style={{color: "red", fontWeight: "bold"}}>{error}</p>}
           </form>
+          {token && (
+            <button className="wizkid-edit-button-set-employement" onClick={handleSubmitChangeEmployement}>
+              {employmentButtonText}
+            </button>
+          )}
```
Comment (cjbreg) [inline src/pages/WizkidEdit.js]: Maybe a better solution is to not even allow a "guest"  on the page at all by adding a check on the router level

File: src/pages/WizkidDetails.js [hunk 4]
```diff
@@ -25,8 +25,12 @@ function WizkidDetails() {
         <WizkidElementContainer>
           <p className="wizkid-details-name">{found.name}</p>
           <p className="wizkid-details-position">{found.position}</p>
-          <p className="wizkid-details-phone-email">{found.phone}</p>
-          <p className="wizkid-details-phone-email">{found.email}</p>
+          {token && (
```
Comment (cjbreg) [inline src/pages/WizkidDetails.js:28]: Might be interesting to have a isLoggedIn value or something, is easier to read/understand if you implement it somewhere

File: src/pages/WizkidEdit.js (hunk 3 above)
Comment (RobertAcsinte) [reply src/pages/WizkidEdit.js]: You mean something like this?
```
  return (
    <div>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/details/:id" element={<WizkidDetails />} />
        {token && <Route path="/edit/:id" element={<WizkidEdit />} />}
        <Route path="/login" element={<Login />} />
      </Routes>
    </div>

  );
```

And then hide the edit button in the function that shows the details?

**Pull request merged**